app/master/cluster_master.py

```diff
@@ -151,8 +161,7 @@ def _disconnect_slave(self, slave):
         """
         # Mark slave dead. We do not remove it from the list of all slaves. We also do not remove it from idle_slaves;
         # that will happen during slave allocation.
-        slave.set_is_alive(False)
-        slave.current_build_id = None
+        slave.disconnect()
```

Comment:
Although we might want to rename the method on the Slave object so it's clear that this is not actually sending the slave any requests or telling the slave to do anything. Maybe something like `mark_disconnected`?